core/chain_indexer.go

```diff
@@ -202,9 +201,7 @@ func (c *ChainIndexer) eventLoop(currentHeader *types.Header, ch chan ChainEvent
 			}
 			header := ev.Block.Header()
 			if header.ParentHash != prevHash {
-				if h := FindCommonAncestor(c.chainDb, prevHeader, header); h != nil {
-					c.newHead(h.Number.Uint64(), true)
-				}
+				c.newHead(FindCommonAncestor(c.chainDb, prevHeader, header).Number.Uint64(), true)
```

Comment:
This is not correct, it will panic after light client checkpoint syncing. I think that is the only case where you don't have a common ancestor in the header chain. We can think about a nicer workaround for this corner case but this one will definitely break it.